File: not_solved/9663.py
# 굳이 둘째 줄 이상을 더 탐색할 필요는 없음
def n_queen(N, r, Q_pos):
    global cnt
    for i in range(N):
        is_possible = True
        for r_q, c_q in Q_pos:
            if (r_q - c_q) == r - i or (r_q + c_q) == r + i or r == r_q or i == c_q:
                is_possible = False
                break
                
        if is_possible:
            Q_pos += [(r, i)]
            if r == N-1:
                cnt += 1
            else:
                n_queen(N, r+1, Q_pos)

            del Q_pos[-1]


N = int(input())
cnt = 0
n_queen(N, 0, [])
print(cnt)

File: not_solved/9663.c
# include <stdio.h>

int cnt = 0;

void N_Queen(char* Q_pos, int N, int r){
    int r_q;
    int c_q;
    char is_possible = 1;
    for(int i = 0; i < N; i ++){
        is_possible = 1;
        for(int q = 0; q < r; q++){
            r_q = q;
            c_q = Q_pos[q];

            if ((r_q == r) || (c_q == i) || (r_q + c_q == r + i) || (r_q - c_q == r - i)){
                is_possible = 0;
                break;
            }
        }
        if (is_possible) {
            Q_pos[r] = i;
            if (r == N-1) cnt++;
            else N_Queen(Q_pos, N, r+1);
            Q_pos[r] = 0;
        }

    }
}

int main(){
    int N;
    scanf("%d", &N);
    char Q_pos[14] = {0,};

    N_Queen(Q_pos, N, 0);

    printf("%d\n", cnt);
    return 0;
}